src/Hosting/TestHost/src/WebHostBuilderExtensions.cs

```diff
@@ -33,9 +33,17 @@ public static IWebHostBuilder ConfigureTestServices(this IWebHostBuilder webHost
                 throw new ArgumentNullException(nameof(servicesConfiguration));
             }
 
-            webHostBuilder.ConfigureServices(
-                s => s.AddSingleton<IStartupConfigureServicesFilter>(
-                    new ConfigureTestServicesStartupConfigureServicesFilter(servicesConfiguration)));
+            if (webHostBuilder.GetType().Name.Equals("GenericWebHostBuilder"))
+            {
+                // Generic host doesn't need to do anything special here since there's only one container.
+                webHostBuilder.ConfigureServices(servicesConfiguration);
```

Comment:
We need to make sure there is a way for these to run AFTER Startup.ConfigureServices otherwise they wont be able to override services configured in Startup.ConfigureServices in a definitive way. 

/cc @davidfowl.
My suggestion is for GenericWebHostBuilder to have a feature for it that these can be queued and applied on along with the ConfigureServices defined by the user in startup directly.

The main point is to keep the app untouched while allowing the service to be configured for tests within the tests assembly.

Reply:
They already do run after Startup.ConfigureServices, I'd updated the tests to prove that. UseStartup registers in the ConfigureServices list immediately so ConfigureTestServices works fine so long as it's called after UseStartup (which it always should be anyways).